src/main/java/teammates/storage/api/InstructorsDb.java

```diff
@@ -191,6 +192,10 @@ public InstructorAttributes updateInstructorByGoogleId(InstructorAttributes.Upda
         InstructorAttributes newAttributes = makeAttributes(instructor);
         newAttributes.update(updateOptions);
 
+        if (!newAttributes.role.equals("Custom")) {
+            newAttributes.privileges = new InstructorPrivileges(newAttributes.role);
```

Comment:
This is not the right place to do it. The one that should be changed is the `updateOptions` being passed.